### PR TITLE
refs [TASK][33693] init detail ingredient detail

### PR DESCRIPTION
**Pull request by phong10101998**
## Related Tickets
- [#Task 33693: Content](https://edu-redmine.sun-asterisk.vn/issues/33693)
## WHAT
- Truyền name và link ảnh sang Detail Ingredient Fragment
- Get data theo tên Ingredient từ API
- Tạo UI
- Đổ dữ liệu lên View
## Evidence (Screenshot or Video)
image

## Review Checklist

Category | View Point | Description | Expected Reviewer Answer | Self review | Reviewer2 (name)
--- | --- | --- | --- | --- | ---
Conventions | Does the code follow Sun* coding style and coding conventions? | https://github.com/framgia/coding-standards/tree/master/eng/android | YES |<li>- [ ] yes</li>|<li>- [ ] yes</li>  
Redmine | Does the ticket follow Sun* Redmine working process?  | https://github.com/framgia/Training-Guideline/blob/master/WorkingProcess/redmine/redmine.md| YES |<li>- [ ] yes</li>|<li>- [ ] yes</li>  
Documentation | Is there any incomplete code? If so, should it be removed or flagged with a suitable marker like ‘TODO’? |  | YES |<li>- [ ] yes</li>|<li>- [ ] yes</li>

### CODE REVIEW

File: app/src/main/java/com/sun/cookbook/screen/ingredient/IngredientDetailFragment.kt [hunk 1]
```diff
@@ -5,14 +5,91 @@ import androidx.fragment.app.Fragment
...
+    companion object {
+
+        private const val BUNDLE_NAME_INGREDIENT = "BUNDLE_NAME_INGREDIENT"
+        private const val BUNDLE_IMAGE_INGREDIENT = "BUNDLE_IMAGE_INGREDIENT"
+
+        fun newInstance(ingredient: Ingredient) = IngredientDetailFragment().apply {
+            arguments = Bundle().apply {
+                putString(BUNDLE_NAME_INGREDIENT, ingredient.name)
+                putString(BUNDLE_IMAGE_INGREDIENT, ingredient.image)
+            }
```
Comment (huytq-2422) [inline app/src/main/java/com/sun/cookbook/screen/ingredient/IngredientDetailFragment.kt]:
```suggestion
            arguments = bundleOf(
                BUNDLE_NAME_INGREDIENT to ingredient.name
                BUNDLE_IMAGE_INGREDIENT to ingredient.image
            )
```

File: app/src/main/res/layout/fragment_ingredient_detail.xml [hunk 2]
```diff
@@ -1,7 +1,75 @@
...
+            app:layout_constraintStart_toStartOf="parent"
+            app:layout_constraintTop_toTopOf="parent" />
+
+        <androidx.cardview.widget.CardView
+            android:id="@+id/cardCircleImage"
+            android:layout_width="0dp"
+            android:layout_height="0dp"
+            android:layout_centerHorizontal="true"
+            android:layout_marginTop="@dimen/dp_20"
+            android:elevation="12dp"
```
Comment (huytq-2422) [inline app/src/main/res/layout/fragment_ingredient_detail.xml]: .

File: app/src/main/res/layout/fragment_ingredient_detail.xml [hunk 3]
```diff
@@ -1,7 +1,75 @@
...
+        <androidx.cardview.widget.CardView
+            android:id="@+id/cardCircleImage"
+            android:layout_width="0dp"
+            android:layout_height="0dp"
+            android:layout_centerHorizontal="true"
+            android:layout_marginTop="@dimen/dp_20"
+            android:elevation="@dimen/dp_12"
+            android:innerRadius="0dp"
+            android:shape="ring"
+            app:cardCornerRadius="100dp"
```
Comment (tuanlvt-1117) [inline app/src/main/res/layout/fragment_ingredient_detail.xml]: harrd code

File: app/src/main/res/layout/fragment_ingredient_detail.xml [hunk 4]
```diff
@@ -1,7 +1,75 @@
...
+            app:layout_constraintDimensionRatio="1:1"
+            app:layout_constraintEnd_toEndOf="parent"
+            app:layout_constraintStart_toStartOf="parent"
+            app:layout_constraintTop_toTopOf="parent"
+            app:layout_constraintWidth_percent="0.45">
+
+            <ImageView
+                android:id="@+id/imageViewIngredient"
```
Comment (tuanlvt-1117) [inline app/src/main/res/layout/fragment_ingredient_detail.xml]: dùng 1 format đặt tên rõ ràng e. img hoặc imageView chứ không phải mỗi nơi đặt 1 kiểu

File: app/src/main/res/layout/fragment_ingredient_detail.xml [hunk 5]
```diff
@@ -1,7 +1,74 @@
...
+            android:innerRadius="0dp"
+            app:cardCornerRadius="@dimen/dp_100"
+            app:layout_constraintDimensionRatio="1:1"
+            app:layout_constraintEnd_toEndOf="parent"
+            app:layout_constraintStart_toStartOf="parent"
+            app:layout_constraintTop_toTopOf="parent"
+            app:layout_constraintWidth_percent="0.45">
+
+            <ImageView
+                android:id="@+id/imageIngredient"
```
Comment (tuanlvt-1117) [inline app/src/main/res/layout/fragment_ingredient_detail.xml]: check lại

**Review by tuanlvt-1117 — APPROVED** [review]

**Pull request merged**